aten/src/ATen/native/SoftMax.cpp

```diff
@@ -203,7 +203,10 @@ void host_softmax(
           // update output
           for (const auto d : c10::irange(dim_size)) {
             // LogSoftMax and MaskedSoftMax should not both be true
-            if (LogSoftMax) {
+            if (MaskedSoftMax && !mask_data[d * dim_stride]) {
+              output_data[d * dim_stride] = 0;
```

Comment:
This is not needed, since when (MaskedSoftMax && !mask_data[d * dim_stride]) is true, output_data[...] is always 0, this is assured at line 186 and 192.

Reply:
Yeah you're right - this was originally used to guarantee that if `tmpsum=0` -- i.e. if the entire row is masked -- then we would still get 0's. But we've updated the behavior to say that it *may* return all 0's, so I'll delete this